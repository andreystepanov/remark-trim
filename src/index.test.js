import trim from './'
import remark from 'remark'

// prettier-ignore
const data = `
# Title

 \` code \`

 \` code  with 2  spaces \`

\`\xa0\xa0\xa0\xa0const param = [  1, 2,   3 ]\xa0\xa0\xa0\xa0\`

\` const param = false  \`

\`\`\`
const block =  true
\`\`\`

 Paragraph with  ** bold **  text .

   Paragraph with  ** * bold italic * **  text  :

Paragraph    with [  link with  spaces    ]( https://link.com )

Paragraph     with image ![  caption with  spaces    ]( https://images.com/demo.png  )

Paragraph with  * italic *  text.

-   List 1
-  List   2
`

function parse(data, options) {
  return remark()
    .use(trim)
    .use({ settings: { fences: true } })
    .processSync(data)
    .toString()
}

test('defined', () => {
  expect(trim).toBeDefined()
  expect(typeof trim).toBe('function')
})

test('trims all unnecessary spaces', () => {
  const md = parse(data)
  expect(md).toMatchSnapshot()
})
