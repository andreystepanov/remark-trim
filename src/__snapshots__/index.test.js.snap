// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trims all unnecessary spaces 1`] = `
"# Title

\`code\`

\`code  with 2  spaces\`

\`    const param = [  1, 2,   3 ]    \`

\`const param = false\`

\`\`\`
const block =  true
\`\`\`

Paragraph with **bold** text.

Paragraph with **_bold italic_** text:

Paragraph with [link with spaces](https://link.com)

Paragraph with image ![caption with spaces](https://images.com/demo.png)

Paragraph with _italic_ text.

-   List 1
-   List 2
"
`;
